pkg/scheduler/util/utils_test.go

```diff
@@ -122,55 +124,104 @@ func TestMoreImportantPod(t *testing.T) {
 	}
 }
 
-func TestRemoveNominatedNodeName(t *testing.T) {
+func TestClearNominatedNodeName(t *testing.T) {
+	t.Parallel()
+	statusErr := &apierrors.StatusError{
+		ErrStatus: metav1.Status{Reason: metav1.StatusReasonUnknown},
+	}
+
 	tests := []struct {
-		name                     string
-		currentNominatedNodeName string
-		newNominatedNodeName     string
-		expectedPatchRequests    int
-		expectedPatchData        string
+		name                  string
+		pods                  []*v1.Pod
+		patchError            error
+		expectedPatchErrors   []error
+		expectedPatchRequests int
+		expectedPatchData     string
 	}{
 		{
-			name:                     "Should make patch request to clear node name",
-			currentNominatedNodeName: "node1",
-			expectedPatchRequests:    1,
-			expectedPatchData:        `{"status":{"nominatedNodeName":null}}`,
+			name: "Should make patch requests to clear node name",
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo2"},
+					Status:     v1.PodStatus{NominatedNodeName: "node2"},
+				},
+			},
+			expectedPatchRequests: 2,
+			expectedPatchData:     `{"status":{"nominatedNodeName":null}}`,
+		},
+		{
+			name:                  "Should not make patch request if nominated node is already cleared",
+			pods:                  []*v1.Pod{},
+			expectedPatchRequests: 0,
 		},
 		{
-			name:                     "Should not make patch request if nominated node is already cleared",
-			currentNominatedNodeName: "",
-			expectedPatchRequests:    0,
+			name: "Should not be patched if NominatedNodeName is empty",
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: ""},
+				},
+			},
+			expectedPatchRequests: 0,
+		},
+		{
+			name: "Should return error if patch fails",
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+				{
+					// In this test, the pod named "err1" returns an error in the patch.
+					ObjectMeta: metav1.ObjectMeta{Name: "err1"},
```

Comment:
Could you fix the `cs.AddReactor` as well

```suggestion
					// In this test, the pod named "err" returns an error in the patch.
					ObjectMeta: metav1.ObjectMeta{Name: "err"},
```